fix: unify epoch display across all DNF5 commands

fhbash opened the pull request:
Always show epoch (including 0) in package NEVRA strings for consistency. Updates Package::to_string(), transaction table, daemon API, and transaction replay.

Fixes inconsistent epoch handling where some commands showed epoch and others didn't.

Closes #1175

kontura commented on include/libdnf5/rpm/package.hpp:
```diff
@@ -146,13 +146,15 @@ class LIBDNF_API Package {
     std::string get_arch() const;
 
     /// @return RPM package EVR (Epoch:Version-Release). If the Epoch is 0, it is omitted from the output.
+    /// @deprecated Use separate components or full NEVRA for consistent epoch display
```
Do we really want to deprecate these?
I would be fine with leaving them on the API.

fhbash replied:
I don't have strong opinion on this, my thought to deprecated is warning API users is better to use full nevra, but I'm good to leave them like it was if it is the case

fhbash replied:
Reverted these changes as suggested.

kontura reviewed (commented):
I like this change but personally I would treat it as an API break.

dcantrell reviewed (changes requested):
I also agree that this constitutes an API break.  Epoch is an odd one in the RPM space.  Many packagers and users do not understand it and that has led to some packages receiving Epoch updates when they should not have.  Most packages have '0' as the Epoch and then you have a number of them with crazy high epoch values.

I think if dnf5 were to start showing epoch all the time it would lead to further misuse of epoch among users and package maintainers.  I would like to see this functionality added to dnf5 as an option you can turn on via the config file, but leave the default as-is.  That brings in the new functionality but does not break the API as we have it now.  Minimal disruption.

dcantrell reviewed (approved):
Looks good to me now, thanks.

kontura commented:
Note also that the CI tests are failing (see the [checks](https://github.com/rpm-software-management/dnf5/pull/2447/checks)), looks like some of that are valid problems, in particular I think there is a problem with replaying transactions. However some other tests will simply need to be updated as they expect nevras without 0 epoch.

The CI is here: https://github.com/rpm-software-management/ci-dnf-stack/

Feel free to ping me if you would like some help with that.

fhbash commented on dnf5daemon-server/package.cpp:
```diff
@@ -329,10 +329,19 @@ std::string package_to_json(const libdnf5::rpm::Package & libdnf_package, const
+                    if (evr_with_epoch.empty()) {
+                        evr_with_epoch = "0";
+                    }
+                    evr_with_epoch += ":" + libdnf_package.get_version() + "-" + libdnf_package.get_release();
```
With this change, I believe we'll always have epoch = 0

fhbash commented on libdnf5-cli/output/transaction_table.cpp:
```diff
@@ -336,7 +336,13 @@ TransactionTable::Impl::Impl(ITransaction & transaction) {
+        if (evr_with_epoch.empty()) {
+            evr_with_epoch = "0";
+        }
+        evr_with_epoch += ":" + pkg->get_version() + "-" + pkg->get_release();
```
See above comment

kontura reviewed (commented):
Thanks.

I think that now https://github.com/rpm-software-management/ci-dnf-stack/pull/1765 needs an update. I believe that all the PR has to do is just a few smaller changes adding the 0 epoch in a couple places in these 4 files: install-package-ursine.feature, transaction-output.feature, replay-errors.feature and replay.feature.

fhbash commented:
/packit test

fhbash committed:
fix: unify epoch display across all DNF5 commands

Always show epoch (including 0) in package NEVRA strings for consistency.
Updates Package::to_string(), transaction table.

Fixes inconsistent epoch handling where some commands showed epoch and others didn't.

Signed-off-by: Fellipe Henrique <me@fhbash.com>

kontura commented:
Thank you!
